frontend/src/lobby/PlayerList.tsx

```diff
@@ -1,74 +1,58 @@
+import { Accessor, Component, For, Show, createSignal } from "solid-js";
 import { TPlayer } from "../types";
-import { sendMessage } from "../App";
-import { setUserNameMsg } from "../SetUserName";
 
-let [changeName, setChangeName] = createSignal<boolean>(false);
-const [container, setContainer] = createSignal<HTMLDivElement>();
+export const [userName, setUserName] = createSignal<string>("");
 
 const PlayerList: Component<{
-  players: TPlayer[];
+  players: Accessor<TPlayer[] | null>;
   pointsKey: "points" | "points_current_round";
   id: string;
 }> = (props) => {
   let id = localStorage.getItem("id");
-  let [userName, setUserName] = createSignal<string>(props.players.find(player =>
-    player[0].id === id
-  )[0].name);
+  setUserName(props.players().find(player => player[0].id === id)[0].name);
 
   return (
     <For
-      each={props.players.sort(
+      each={props.players()?.sort(
         (a: TPlayer, b: TPlayer) =>
           (b[0]?.[props.pointsKey] ?? 0) - (a[0]?.[props.pointsKey] ?? 0) ?? []
       )}
     >
       {(player: TPlayer) => (
-        <>
-           <div class="w-fit">
-            <Show when={!changeName() || player[0].id !== id}
-              fallback={
-                <span onkeyup={(key) => {
-                  if (key.key == "Enter" || key.key == "Escape") {
-                    setChangeName(false)
+        <div class="w-fit">
+          <Show when={player[0].id == props.id} fallback={
+            <>
+              <span class="font-light" onclick={() => {
+                if (player[0].id == props.id) {
+                }
+              }
+              }>
+                {player[0].name.length > 12
+                  ? player[0].name.substring(0, 12)
+                  : player[0].name}
+              </span>
+              :<span class="font-bold"> {player[0][props.pointsKey] ?? 0} </span>
+            </>
+          }>
+            <p>
+              <label for="my-modal-6" >
+                <span class="font-light " onclick={() => {
+                  if (player[0].id == props.id) {
                   }
                 }
                 }>
-                  <input
-                    ref={setContainer}
-                    value={userName()}
-                    onchange={(e) => {
-                      let msg = setUserNameMsg;
-                      setUserName(e.target.value)
-                      setUserNameMsg.args.name = userName();
-                      sendMessage(msg);
-                      localStorage.setItem("username", userName());
-                    }}
-                    onblur={() => setChangeName(false)}
-                  >
-                    {player[0].name.length > 12
-                      ? player[0].name.substring(0, 12)
-                      : player[0].name}
-                    :{" "}
-                  </input>
+                  {player[0].name.length > 12
+                    ? player[0].name.substring(0, 12)
```

Comment:
is 12 inclusive or exclusive? otherwise we should check if its bigger than 11 or change the substring end to 13. i think you  just copied this from me so we also  need to fix that in an other position

Reply:
is inclusiv "123".length == 3

Reply:
I mean substring()